applications/tari_validator_node/src/payload_processor.rs

```diff
@@ -102,6 +101,28 @@ where TTemplateProvider: TemplateProvider<Template = LoadedTemplate>
     }
 }
 
+fn build_package<TTemplateProvider: TemplateProvider<Template = LoadedTemplate>>(
+    template_provider: &TTemplateProvider,
+    template_addresses: HashSet<TemplateAddress>,
+) -> Result<Package, PayloadProcessorError> {
+    let mut builder = Package::builder();
+
+    for addr in template_addresses {
+        let template = template_provider
+            .get_template_module(&addr)
+            .map_err(|err| PayloadProcessorError::FailedToLoadTemplate(err.into()))?;
+        builder.add_template(addr, template);
+    }
+
+    Ok(builder.build())
+}
+
+fn get_auth_token(transaction: &Transaction) -> NonFungibleAddress {
+    let public_key = Ed25519PublicKey::from_bytes(transaction.sender_public_key().as_bytes())
```

Comment:
why not ristretto?

Reply:
Ristretto is the group so thought it better to specify the curve since ristretto is more of a detail. It refers to the Resource more than the actual contents. We could, for example, use a hash of the key and/or part of the entropy of the key for the NFT ID just as long as it's derived from an ED25519 curve. Though happy to use Ristretto as the name, just a decision I made :)

Reply:
The problem with reading an Ed25519 from bytes is that there could be 4 private keys because of the co-factor. I think let's stick to Ristretto for everything for now and we can expand as the need arises